code/src/components/GroceryList.js

```diff
@@ -0,0 +1,41 @@
+/* eslint-disable jsx-a11y/label-has-associated-control */
+import React from 'react';
+import { useSelector, useDispatch } from 'react-redux';
+import items from 'reducers/Item';
+import { Background, Text } from './Styling/GlobalStyles';
+
+const ItemsList = () => {
+  const itemList = useSelector((store) => store.items.items);
+  const dispatch = useDispatch();
+  const onIsBoughtToggle = (id) => {
+    dispatch(items.actions.toggleItem(id));
+  }
+  const onDeleteItemBtnClick = (itemIndex) => {
+    dispatch(items.actions.deleteItem(itemIndex));
+  }
+  return (
+    <Background>
+      <section>
+        {itemList.map((singleItem, index) => {
+          return (
+            <article key={singleItem.id}>
+              <Text>{singleItem.name}</Text>
+              <label>
+                <input
+                  type="checkbox"
+                  checked={singleItem.isBought}
+                  onChange={() => onIsBoughtToggle(singleItem.id)} />
+              </label>
+              <button
+                onClick={() => onDeleteItemBtnClick(index)}
```

Comment:
The delete button does not work,  I think it could be that in reducer you use both mutable and immutable ways of doing it, I compered it with my code and there is no other error that I could see. I chose only mutable way.